### PR TITLE
K-NN refresh blog

### LINKED ISSUE #1247: [Blog] Build K-Nearest Neighbor (k-NN) Similarity Search Engine with Elasticsearch - Refresh
**Describe the blog post your would like to write**
Refreshing the existing blog content and updating the links for the Build K-Nearest Neighbor (k-NN) Similarity Search Engine with Elasticsearch since it is from 2020 and some of the references are dated. Linked here: https://opensearch.org/blog/Building-k-Nearest-Neighbor-(k-NN)-Similarity-Search-Engine-with-Elasticsearch/

**What is the title of the blog post?**
Build K-Nearest Neighbor (k-NN) Similarity Search Engine with Elasticsearch

**Who are the authors?**
Jack Mazanec and Vamshi Vijay Nakkirtha

**What is the proposed posting date?**
1/6/23

### PR DESCRIPTION
**Pull request by kolchfa-aws**
### Description
Refreshes an old k-NN blog
 
### Issues Resolved
Fixes #1247 

### Check List
- [x ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

File: _posts/2023-01-09-knn-similarity-refresh.markdown [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+date: 2023-01-09 00:00:00 -0700
+categories:
+ - technical-post
+meta_keywords: "approximate k-NN search, k-nearest neighbor plugin, k-NN plugin, ANN similarity search solution"
+meta_description: "Learn how the OpenSearch approximate k-NN search solution enables you to build a scalable, reliable, and distributed framework for similarity searches" 
+
+excerpt: Since the OpenSearch Project introduced the k-nearest neighbor plugin in 2019, it has supported both exact and approximate k-NN search. The approximate k-NN search method is more efficient for large datasets with high dimensionality because it reduces the cardinality of searchable vectors. This approach is superior in speed at the cost of a slight reduction in accuracy. 
+---
+
+# Build k-Nearest Neighbor (k-NN) Similarity Search Engine with OpenSearch
```
Comment (krisfreedain) [inline _posts/2023-01-09-knn-similarity-refresh.markdown:18]: Please remove this title  - it appears as duplicative

File: _posts/2023-01-09-knn-similarity-refresh.markdown [hunk 2]
```diff
@@ -0,0 +1,215 @@
+---
+layout: post
+title:  "Build k-Nearest Neighbor (k-NN) Similarity Search Engine with OpenSearch"
+authors:
+- vamshin
+- laijiang
+- csw
+- jmazane
+date: 2023-01-09 00:00:00 -0700
```
Comment (krisfreedain) [inline _posts/2023-01-09-knn-similarity-refresh.markdown:9]: I don't believe we typically update the date here - please confirm @pajuric - and please set this back to the original blog date. This isn't a new blog, it's a refresh.

**Review by krisfreedain — CHANGES_REQUESTED** [review]
@kolchfa-aws  a couple changes needed. CC @pajuric

File: _posts/2023-01-09-knn-similarity-refresh.markdown (hunk 2 above)
Comment (pajuric) [reply _posts/2023-01-09-knn-similarity-refresh.markdown:9]: Correct - we leave the original publish date for the blog, which was Mon, Apr 06, 2020.

File: _posts/2023-01-09-knn-similarity-refresh.markdown (hunk 1 above)
Comment (kolchfa-aws) [reply _posts/2023-01-09-knn-similarity-refresh.markdown]: Done

File: _posts/2023-01-09-knn-similarity-refresh.markdown (hunk 2 above)
Comment (kolchfa-aws) [reply _posts/2023-01-09-knn-similarity-refresh.markdown]: Put the original date, but named the blog today's date.

**Comment by krisfreedain** [comment]
Just noticed this - curious why the filename was also changed to today's date - this will end up making it a duplicate of the 2020-04-06 post when it should be a refresh of that post/file.

**Comment by kolchfa-aws** [comment]
@krisfreedain I changed the file name to the original. I also changed the front matter of the file - could you look it over and see if it's OK, or does it need to be exactly like the original? Thanks.

**Comment by pajuric** [comment]
Checking status on this. Do we need to redo the PR to get this one moving?

**Review by krisfreedain — APPROVED** [review]
thank you @kolchfa-aws

**Pull request merged**